src/main/java/duke/DateTimeParser.java

```diff
@@ -0,0 +1,41 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * A class representing a parser that returns a date.
+ */
+public class DateTimeParser extends Parser<LocalDate> {
```

Comment:
I like how you have your own `Parser` with generic, I think it makes the code neat